### PR TITLE
add typehints of part 3 in mmdet/models/backbones/csp_darknet.py

### PR DESCRIPTION
**Pull request by hjy94wo777**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Add typehints in mmdet/models/backbones/csp_darknet.py

## Modification

Add typehints in mmdet/models/backbones/csp_darknet.py

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by ZwwWayne** [comment]
Hi @hjy94wo777 

Thanks for your kind PR. It seems that CLA is not signed. Could you sign the CLA so that eventually we could merge this PR after review? You can check the contents and follow the instruction in the communication box shown as below
image

File: mmdet/models/backbones/csp_darknet.py [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
+from typing import Optional, Sequence, Union
 
 import torch
 import torch.nn as nn
 from mmcv.cnn import ConvModule, DepthwiseSeparableConvModule
 from mmcv.runner import BaseModule
+from torch import Tensor
 from torch.nn.modules.batchnorm import _BatchNorm
 
 from ..builder import BACKBONES
```
Comment (BIGWangYuDong) [inline mmdet/models/backbones/csp_darknet.py:12]: from mmdet.utils import ConfigType, OptConfigType

File: mmdet/models/backbones/csp_darknet.py [hunk 2]
```diff
@@ -28,13 +30,13 @@ class Focus(nn.Module):
...
-                 kernel_size=1,
-                 stride=1,
-                 conv_cfg=None,
-                 norm_cfg=dict(type='BN', momentum=0.03, eps=0.001),
-                 act_cfg=dict(type='Swish')):
+                 in_channels: int,
+                 out_channels: int,
+                 kernel_size: int = 1,
+                 stride: int = 1,
+                 conv_cfg: Optional[dict] = None,
```
Comment (BIGWangYuDong) [inline mmdet/models/backbones/csp_darknet.py:37]: OptConfigType

File: mmdet/models/backbones/csp_darknet.py [hunk 3]
```diff
@@ -28,13 +30,13 @@ class Focus(nn.Module):
...
+                 norm_cfg: dict = dict(type='BN', momentum=0.03, eps=0.001),
```
Comment (BIGWangYuDong) [inline mmdet/models/backbones/csp_darknet.py:38]: ConfigType

File: mmdet/models/backbones/csp_darknet.py [hunk 4]
```diff
@@ -28,13 +30,13 @@ class Focus(nn.Module):
...
+                 act_cfg: dict = dict(type='Swish')):
```
Comment (BIGWangYuDong) [inline mmdet/models/backbones/csp_darknet.py:39]: ConfigType

Comment (BIGWangYuDong) [inline mmdet/models/backbones/csp_darknet.py:39]: -> None

File: mmdet/models/backbones/csp_darknet.py [hunk 5]
```diff
@@ -46,7 +48,7 @@ def __init__(self,
             norm_cfg=norm_cfg,
             act_cfg=act_cfg)
 
-    def forward(self, x):
+    def forward(self, x: Tensor) -> Tensor:
         # shape of x (b,c,w,h) -> y(b,4c,w/2,h/2)
```
Comment (BIGWangYuDong) [inline mmdet/models/backbones/csp_darknet.py:52]: add doc string here

File: mmdet/models/backbones/csp_darknet.py [hunk 6]
```diff
@@ -83,13 +85,13 @@ class SPPBottleneck(BaseModule):
...
-                 out_channels,
-                 kernel_sizes=(5, 9, 13),
-                 conv_cfg=None,
-                 norm_cfg=dict(type='BN', momentum=0.03, eps=0.001),
-                 act_cfg=dict(type='Swish'),
-                 init_cfg=None):
+                 in_channels: int,
+                 out_channels: int,
+                 kernel_sizes: Sequence[int] = (5, 9, 13),
+                 conv_cfg: Optional[dict] = None,
```
Comment (BIGWangYuDong) [inline mmdet/models/backbones/csp_darknet.py:91]: Same

**Comment by BIGWangYuDong** [comment]
Thanks for your PR, you can follow https://github.com/open-mmlab/mmdetection/blob/3.x/mmdet/models/dense_heads/atss_head.py to update your typehint. 

Especially, dict or Option[dict] can use `ConfigType` or `OptConfigType` in `__init__`

**Comment by BIGWangYuDong** [comment]
CI seems failed, should check whether got error in this PR

**Comment by hjy94wo777** [comment]
> Hi @hjy94wo777
> 
> Thanks for your kind PR. It seems that CLA is not signed. Could you sign the CLA so that eventually we could merge this PR after review? You can check the contents and follow the instruction in the communication box shown as below image
Sorry, I forgot about that. I have signed the CLA now.

**Pull request closed without merging**